deepchem/feat/smiles_tokenizer.py

```diff
@@ -0,0 +1,183 @@
+# Requriments - transformers, tokenizers
+# Right now, the Smiles Tokenizer uses an exiesting vocab file from rxnfp that is fairly comprehensive and from the USPTO dataset.
+# The vocab may be expanded in the near future
+
+import collections
+import logging
+import os
+import re
+import numpy as np
+import pkg_resources
+from typing import List
+from transformers import BertTokenizer
+
+# export
+SMI_REGEX_PATTERN = r"(\[[^\]]+]|Br?|Cl?|N|O|S|P|F|I|b|c|n|o|s|p|\(|\)|\.|=|#|-|\+|\\|\/|:|~|@|\?|>>?|\*|\$|\%[0-9]{2}|[0-9])"
+
+def get_default_tokenizer():
+    default_vocab_path = (
+        pkg_resources.resource_filename(
+            "chemberta",
+            "tokenizers/vocab.txt"
+        )
+    )
+    return SmilesTokenizer(default_vocab_path)
+
+class SmilesTokenizer(BertTokenizer):
+    r"""
+    Constructs a SmilesTokenizer.
+    Bulk of code is from https://github.com/huggingface/transformers and https://github.com/rxn4chemistry/rxnfp
+    Args:
```

Comment:
Could you add a usage example here showing how to invoke the tokenizer?

Reply:
will do, i believe the test has an example use-case we can port over here

Reply:
done